dash.el

```diff
@@ -921,9 +921,11 @@ See also: `-drop'"
   "Rotate LIST N places to the right.  With N negative, rotate to the left.
 The time complexity is O(n)."
   (declare (pure t) (side-effect-free t))
-  (if (> n 0)
-      (append (last list n) (butlast list n))
-    (append (-drop (- n) list) (-take (- n) list))))
+  (and list
+       (let* ((list-len (length list))
+              (n-mod-len (mod n list-len))
```

Comment:
It's nice here that it gets rid of the special case for negatives as it automatically produces the corresponding positive offset <3